03 Writing Algorithms/03 Securities/99 Asset Classes/02 Equity Options/02 Handling Data/99 Examples.html

```diff
@@ -0,0 +1,200 @@
+<h4>Example 1: Get Mid Price For Individual Contracts</h4>
+<p>This example shows how to handle <code>QuoteBar</code> data for shortlisted Equity Option contracts to calculate mid price using bid close and ask close data, while <a href="/docs/v2/writing-algorithms/securities/asset-classes/equity-options/requesting-data/individual-contracts">filter individual option contracts and request data</a> using <code class="csharp">OptionChainProvider.GetOptionContractList</code><code class="python">self.option_chain_provider.get_option_contract_list</code> method for the contracts that expires within the current week. Using mid price, we can examine the market fair value of the Option and compare with model theoretical price.</p>
+<div class="section-example-container">
+	<pre class="csharp">public class EquityOptionHandlingDataAlgorithm : QCAlgorithm
+{
+    private Symbol _spy;
+    private List&lt;Symbol&gt; _contracts = new();
+        
+    public override void Initialize()
+    {
+        // Seed the price with last known price to ensure the underlying price data is available on initial option contract filtering.
+        SetSecurityInitializer((security) =&gt; new FuncSecuritySeeder(GetLastKnownPrices).SeedSecurity(security));
+        // Subscribe to underlying data for ATM calculation using the update underlying price.
+        // Set data normalization mode to raw is required to ensure strike price and underlying price is comparable.
+        _spy = AddEquity("SPY", dataNormalizationMode: DataNormalizationMode.Raw).Symbol;
+
+        // Update the tradable contracts daily before market open since the option contract list provider populate them daily.
+        Schedule.On(
+            DateRules.EveryDay(_spy),
+            TimeRules.At(9, 0),
+            UpdateContracts
+        );
+    }
+
+    private void UpdateContracts()
+    {
+        // Get all contracts that expiring this week to trade with, subscribe to data for trading need.
+        _contracts = OptionChainProvider.GetOptionContractList(_spy, Time)
+            .Where(x =&gt; x.ID.Date &lt;= Expiry.EndOfWeek(Time))
+            .Select(x =&gt; AddOptionContract(x).Symbol)
+            .ToList();
+    }
+
+    public override void OnData(Slice slice)
+    {
+        // Only focus on filtered list of option contracts to trade.
+        foreach (var contract in _contracts)
+        {
+            // Mid price can only be calculated when quote bar data is available.
+            if (slice.QuoteBars.ContainsKey(contract))
```

Comment:
I suggest using TryGetValue since we check for key and extract the object in one line:
```csharp
if (slice.QuoteBars.TryGetValue(contract, out var quoteBar))
{
}
```